Домашнее задание 8 спринта, версия 1

DaniilKosmachev opened the pull request:
Экраны сверстаны (в экране настроек не особо понял, потому сделял LinearLayout - у кнопок серые границы)

DaniilKosmachev committed:
Домашнее задание 8 спринта, версия 1

DaniilKosmachev committed:
Домашнее задание 8 спринта, версия 2 (вместо startActivity на activity_settings, сделал finish())

faritowich commented on app/src/main/java/com/example/playlistmaker/MainActivity.kt:
```diff
@@ -1,11 +1,40 @@
         setContentView(R.layout.activity_main)
+        val button_settings = findViewById<Button>(R.id.setting_window)
+        val button_search = findViewById<Button>(R.id.search_window)
+        val button_library = findViewById<Button>(R.id.library_window)
```
Названия переменных и методов в Котлин и Джаве принято писать camelCase'ом, то есть buttonLibrary вместо button_library. Айдишники в лейаутах да, часто пишутся в snake_case (хотя я и их предпочитаю писать в camelCase), но соответствующие им переменные надо переводить в camelCase

DaniilKosmachev replied:
Да, понял, на ходу после xml написал

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,76 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    tools:context=".MainActivity">
+    android:background="#3772E7"
```
Все цвета должны быть в ресурсах - тут и в других местах такое

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,76 @@
+            android:gravity="start"
+            android:textColor="#FFFFFF"
+            android:textSize="@dimen/main_text_size"/>
+    </FrameLayout>
```
Лишняя вложенность, FrameLayout не нужен

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,76 @@
+        <Button
+            android:id="@+id/search_window"
+            android:layout_width="match_parent"
+            android:layout_height="@dimen/main_button_height"
```
Конкретную высоту надо обязательно поменять на 0dp. Сейчас на очень больших экранах остается место, а на маленьких наоборот кнопки будут не влезать. Тут надо задать 0dp (см теорию) и одинаковый layout_weight - тогда кнопки и так будут равномерно распределяться

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,152 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:foreground="?attr/selectableItemBackground"
+        android:clickable="true"
```
LinearLayout и так по дефолту clickable, если не ошибаюсь, так что можно убрать это везде

DaniilKosmachev replied:
Видимо, нет, тогда не срабатывает foreground

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,152 @@
+        android:clickable="true"
+        android:focusable="true">
+        <TextView
+            android:layout_width="wrap_content"
```
Лучше поменять на 0dp - студия подсвечивает желтым и советует это для улучшения производительности

faritowich commented on app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <dimen name="settings_padding_bottom">18dp</dimen>
+    <dimen name="settings_padding_top">19dp</dimen>
```
Рекомендуется делать размеры View кратными 8 (или 4). Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Иногда эту рекомендацию приходится нарушать, но лучше по возможности ей следовать.
Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

DaniilKosmachev committed:
Домашнее задание 8 спринта, версия 3
1) Поменял названия переменных в MainActivity.kt на camelCase
2) Везде посмотрел, поменял цвета вместо конкретных # на @colors
3) Убрал вложенность лишнюю на activity_main.xml (удалил FrameLayout)
4) В activity_main.xml убрал конкретную высоту, поменяв на 0dp в dimens.xml, layout_weight выставил всем button по 1
5) Убрав clickable у LinerLayout в activity_settings.xml при запуске на эмуляторе пропал градиент (от касания расплывается серый фон по layout), пока оставил так, ибо непонятно, проходит click или нет - если от этого не зависит, то плз, напишите))))
6) В activity_settings.xml layout_width поменял на dimens из пункта 4 (т.е. на 0dp)
7) В dimens.xml поменял кратность padding на 16dp, соответственно, оставил один dimen с name settings_padding_vertical

faritowich commented on app/src/main/res/values/dimens.xml:
```diff
@@ -0,0 +1,10 @@
+    <dimen name="settings_text_size">19sp</dimen>
+    <dimen name="main_text_size">22sp</dimen>
+    <dimen name="main_button_corner_radius">15dp</dimen>
+    <dimen name="main_button_height">0dp</dimen>
```
Если 0dp, то обычно это не выносят в ресурсы, а пишут 0dp в xml-файле лейаута
